spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultSchedulerService.java

```diff
@@ -166,6 +166,15 @@ public void unschedule(String scheduleName) {
 				this.schedulerServiceProperties.getMaxSchedulesReturned());
 	}
 
+	@Override
+	public ScheduleInfo getSchedule(String scheduleName) {
+		List<ScheduleInfo> result = list().stream()
+				.filter(scheduleInfo -> scheduleInfo.getScheduleName().equals(scheduleName))
+				.collect(Collectors.toList());
+		Assert.isTrue(!(result.size() > 1), "more than one schedule was returned for scheduleName, should only be one");
+		return result.size() > 0 ? result.get(0) : null;
```

Comment:
If you have 2 in the result, are you sure the first one is the right one?

Reply:
hmm...   Doesn't the assert handle this case?   if the result set has more than one value it should throw an assert.

Reply:
Ah.  I missed that.